src/hw_mainline.js

```diff
@@ -62,17 +62,26 @@ var readPinMux = function (pin, mode, callback) {
         if (err) {
             mode.err = 'readPinctrl error: ' + err;
             if (debug) winston.debug(mode.err);
-            callback(mode);
+            if (callback.length == 1)
+                callback(mode);
+            else
+                callback(mode.err, data);
         }
         mode = parse.modeFromPinctrl(data, muxRegOffset, 0x44e10800, mode);
-        callback(mode);
+        if (callback.length == 1)
+            callback(mode);
+        else
+            callback(null, mode);
     };
     var tryPinctrl = function (exists) {
         if (exists) {
             fs.readFile(pinctrlFile, 'utf8', readPinctrl);
         } else {
             if (debug) winston.debug('getPinMode(' + pin.key + '): no valid mux data');
-            callback(mode);
+            if (callback.length == 1)
+                callback(mode);
+            else
+                callback('getPinMode(' + pin.key + '): no valid mux data', mode);
```

Comment:
Can we try to get some consistency in the error messages? Can you check if all the others are in this same format?